src/common/dataAdaptors/mediaAdaptors.js

```diff
@@ -46,3 +47,17 @@ export const mediaAdaptor = ({ type, data }) => {
 
   return { ...data, message: 'NOT a media document' }
 }
+
+export const entryForViewing = ({ type, data, flatListOfSpeakers = false }) => {
+  const response = {
+    ...mediaAdaptor({ type, data }),
+    ...audienceForViewing({ item: data }),
+  }
+
+  if (flatListOfSpeakers) {
+    const speakersFlatList = response?.speakers?.map((speaker) => speaker?.id)
+    response.speakers = speakersFlatList
```

Comment:
Best to avoid this type of mutation in data adaptors as it can lead to hard to detect bugs. Make use of the `objectsToIdsAdaptor` in this instance.


Also, you could always return a flat list of speakers in addition to the default structure. That way regardless of where the adaptor is being used the consumer would have access to both formats. There would no longer be a need to pass a flat list boolean and we would no longer be mutating the `speakers` returned by mediaAdaptor.

Reply:
yes, makes sense and adding both lists is a great idea.